typing/jkind_axis.ml

```diff
@@ -200,21 +174,34 @@ module Axis = struct
     | Modal Contention -> "contention"
     | Nonmodal Externality -> "externality"
     | Nonmodal Nullability -> "nullability"
+
+  let is_deep (type a) : a t -> bool = function
```

Comment:
Maybe rename this to `is_modal`? I.e., our modes are by default deep, and given the current axes as samples, the distinction seems to be between modal vs nonmodal? Do we have imaginary axes violating that?

This is also a good place to insert a CR saying that `Externality` is in fact `Modal` and will be made so in the future.